client/src/components/Header/components/NavLinks.js

```diff
@@ -1,23 +1,48 @@
 import React from 'react';
 import { Link } from '../../helpers';
 
-export function NavLinks() {
-  return (
-    <div className='main-nav-group'>
-      <ul className={'nav-list display-flex'} role='menu'>
-        <li className='nav-theme' role='menuitem'>
-          <Link to='/learn'>Projects</Link>
-        </li>
-        <li className='nav-theme' role='menuitem'>
-          <Link to='/'>Light</Link>
-        </li>
-        <li className='nav-portfolio' role='menuitem'>
-          <Link to='/portfolio'>Portfolio</Link>
-        </li>
-      </ul>
-    </div>
-  );
+class NavLinks extends React.Component {
+  state = {
+    theme: typeof window !== `undefined` ? window.__theme : null
+  };
+
+  componentDidMount() {
+    window.__onThemeChange = theme => {
+      this.setState({ theme: theme });
+    };
+  }
+
+  componentWillUnmount() {
+    window.__onThemeChange = () => {};
+  }
+
+  render() {
+    return (
+      <div className='main-nav-group'>
+        <ul className={'nav-list display-flex'} role='menu'>
+          <li
+            className='nav-theme'
+            onClick={() =>
+              window.__setPreferredTheme(
+                window.__theme === 'dark' ? 'light' : 'dark'
+              )
+            }
+            role='menuitem'
+          >
+            {this.state.theme === 'dark' ? 'light' : 'dark'}
```

Comment:
This causes a problem with hydration, since state isn't something known ahead of time.  Sorry, didn't spot that until I built the client.

One way around is to set the content with css.  It's a bit hacky, though, so I'll see if I can find a better way.

Reply:
In overreacted.io, the solution was just to add the button to the dom only during hydration.  So, the default render just has an empty space.  As long as both `light`, `dark` and the empty space are all the same size, that should work.

There's a brief instant where the button isn't there, but it's not particularly noticeable.